ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -68,14 +68,18 @@ void log_external_operation(
 
 template <typename T>
 Tensor create_owned_tensor(T* data_ptr, const ttnn::TensorSpec& tensor_spec) {
+    if (tensor_spec.memory_config().is_sharded()) {
+        TT_FATAL(
+            tensor_spec.memory_config().shard_spec.has_value(),
```

Comment:
Perhaps include the condition from the `if` here directly: `TT_FATAL(!tensor_spec.memory_config().is_sharded() or tensor_spec.memory_config().shard_spec.has_value(), ...)` (longer assertion but an extra branch is a bit worse)